interpreter/binary/encode.ml

```diff
@@ -138,8 +145,8 @@ let encode m =
     let var x = vu32 x.it
 
     let memop x {align; offset; _} =
-      let flags = if x.it = 0l then align else align + 0x40 in
-      vu32 (Int32.of_int flags);
+      let flags = bit 6 (x.it <> 0l) (bits 0 5 align 0l) in
```

Comment:
The original code was easier to understand for me. It was additionally confusing for me that there is another `bits` function in decode.ml that does the opposite direction, i.e. extracting bits.

Could you not just write
`let flags = logor (logand align 0x3f) (shift_left x.it, 6)`?

Reply:
Okay, your comments are fair enough, I removed the bit abstractions again and went back to direct bit fiddling, avoiding `+` as you suggested earlier.

> Could you not just write `let flags = logor (logand align 0x3f) (shift_left x.it, 6)`?

Not quite, x.it is a 32bit number, not the single flag bit.